docs/CONTRIBUTING.md

```diff
@@ -1,11 +1,327 @@
+# Contributing to OSTree
+
+The following guide is about OSTree forking, building, adding a command, testing the command, and submitting the change.
+
+- [Getting Started](#getting-started)
+- [Building OSTree](#building-ostree)
+    - [Install Build Dependencies](#install-build-dependencies)
+    - [OSTree Build Commands](#ostree-build-commands)
+- [Testing a Build](#testing-a-build)
+    - [Testing in a Container](#testing-in-a-container)
+    - [Testing in a Virtual Machine](#testing-in-a-virtual-machine)
+- [Tutorial: Adding a basic builtin command to OSTree](#tutorial-adding-a-basic-builtin-command-to-ostree)
+    - [Modifying OSTree](#modifying-ostree)
+    - [OSTree tests](#ostree-tests)
+    - [Submitting a patch](#submitting-a-patch)
+    - [Commit message style](#commit-message-style)
+    - [Returning workflow](#returning-workflow)
+
+---
+
+## [Getting Started](#getting-started)
+
+Fork https://github.com/ostreedev/ostree, the run the following commands.
+
+```bash
+$ git clone https://github.com/<username>/ostree
+$ git remote add upstream https://github.com/ostreedev/ostree
+$ git checkout master
+$ git branch --set-upstream-to=upstream/master master
+```
+Make a branch from master for your patch.
+```bash
+$ git checkout -b <name-of-branch>
+$ git branch --set-upstream-to=upstream/master <name-of-branch>
+```
+
+## [Building OSTree](#building-ostree)
+
+### [Install Build Dependencies](#install-build-dependencies)
+
+```bash
+$ sudo dnf builddep ostree
+$ sudo dnf install @buildsys-build
```

Comment:
I'd phrase this more like "Use distribution build dependencies (if possible)".  Something like:

> For distributions where ostree is already packaged, it's often easy to just use the package manager.  On Fedora: `sudo dnf install @buildsys-build dnf-plugins-core && dnf builddep ostree`.  On CentOS7, `yum-builddep`.  Debian: `apt builddep ostree`, etc.

I'd imagine there's a generic page about this somewhere.

Reply:
Thanks. Made amendments this section.